### ISSUE TITLE
ESLint configuration in  **** eslintrc is invalid:    - Unexpected top-level property "react/sort-comp".

### ISSUE THREAD

**Issue by myNameCao** [eslint/eslint#12691]
"eslint": "^5.16.0",

***.lintstagedrc***
```shell
"src/**/*.{js,jsx}": "eslint"
```

***.eslintrc***

```shell
{
  "extends": ["taro"],
  "settings": {
    "react": {
      "version": "detect"
    }
  },
  "react/sort-comp": [1, {
    "order": [
      "constructor",
      "lifecycle",
      "everything-else",
      "render"
    ]
  }],
  "rules": {
    "no-unused-vars": ["error", { "varsIgnorePattern": "Taro" }],
    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx", ".tsx"] }]
  },
  
    "parser": "babel-eslint"
  }


````

**Comment by g-plane** [comment]
Hi, thanks for this issue.

The item `react/sort-comp` should be put under `rules` object. Hope helps.

**Comment by myNameCao** [comment]
hi thanks  for your  answer

if  i put The item react/sort -comp  under  ```rules```  object . eslint  warn  ```config should be placed after componentWillUnmounteslint(react/sort-comp)```  i think  the code about rules doesn't work
